client/js/views/Home.vue

```diff
@@ -119,8 +119,18 @@ export default {
       return this.$store.state.gameList.games;
     },
   },
-  mounted() {
-    this.$store.dispatch('requestGameList');
+  async mounted() {
+    // Leave any existing lobbies before getting the game list
+    // Need to make sure we do these in order so the order of operations is:
+    //     leave -> list
+    try {
+      await this.$store.dispatch('requestLeaveLobby');
+    } catch (err) {
+      // A user may not actually be in a lobby, so we need to
+      // swallow this error so the component doesn't error out
+      // TODO: Improve this logic with an isInLobby helper
+    }
```

Comment:
This is a _little_ weird, but I wanted to make sure I called out why I'm doing this. We need to await the calls to make sure we do them in order. We also need to make sure we swallow any error from `requestLeaveLobby` because the user may not be in a lobby at the moment.

Any error thrown here would break the component's mounting method.

Reply:
Why do they need to be in order? Is it just for the different error handling? Not a big deal but from a performance standpoint this seems like a potentially reasonable use-case for `Promise.all()`, with the caveat that it could theoretically result in joining a game super quickly and having the requestLeaveLobby fire late and remove you from the game you _just_ joined